pkg/controllerhelpers/scylla.go

```diff
@@ -15,46 +16,78 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/errors"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	corev1schedulinghelpers "k8s.io/component-helpers/scheduling/corev1"
 	"k8s.io/component-helpers/scheduling/corev1/nodeaffinity"
 )
 
-func GetScyllaIPFromService(svc *corev1.Service) (string, error) {
-	if svc.Spec.Type != corev1.ServiceTypeClusterIP {
-		return "", fmt.Errorf("service %s is of type %q instead of %q", naming.ObjRef(svc), svc.Spec.Type, corev1.ServiceTypeClusterIP)
+func GetScyllaHost(ctx context.Context, sc *scyllav1.ScyllaCluster, coreV1Client corev1client.CoreV1Interface, svc *corev1.Service) (string, error) {
```

Comment:
> coreV1Client corev1client.CoreV1Interface

why is this using a live client and not a lister?
(applies globally)

Reply:
I figured I'll just make it take a pod as argument - easier to test